### PR TITLE
Add user-guide for VMI with Istio

### PR DESCRIPTION
**Pull request by rhrazdil**
Signed-off-by: Radim Hrazdil <rhrazdil@redhat.com>

### CODE REVIEW

File: docs/virtual_machines/istio_service_mesh.md
Comment (rhrazdil) [inline docs/virtual_machines/istio_service_mesh.md:71]: @phoracek Not sure if this section is really helpful, wdyt?

File: docs/virtual_machines/istio_service_mesh.md
Comment (rhrazdil) [inline docs/virtual_machines/istio_service_mesh.md:82]: @phoracek I thought about adding a `Troubleshooting` section with possible common mistakes, but I'm not sure if it fits into our user-guide, wdyt?

File: docs/virtual_machines/istio_service_mesh.md
Comment (phoracek) [reply docs/virtual_machines/istio_service_mesh.md:84]: I am all for verification steps. It would help bisect any bug that may happen here. +1

File: docs/virtual_machines/istio_service_mesh.md
Comment (phoracek) [reply docs/virtual_machines/istio_service_mesh.md:92]: I am personally in favor of troubleshooting section. It is not as good as bulletproof system but well it is better than running around clueless on what to do. +++

File: docs/virtual_machines/istio_service_mesh.md
Comment (phoracek) [inline docs/virtual_machines/istio_service_mesh.md:10]:
```suggestion
- Istio uses a [list of ports](https://istio.io/latest/docs/ops/deployment/requirements/#ports-used-by-istio) for its own purposes, these ports must not
```

File: docs/virtual_machines/istio_service_mesh.md
Comment (phoracek) [inline docs/virtual_machines/istio_service_mesh.md:19]: Is not this a part of the prerequisite? If it is, maybe it should be a part of troubleshooting if you consider this a common mistake? I won't fight for this

File: docs/virtual_machines/istio_service_mesh.md
Comment (phoracek) [inline docs/virtual_machines/istio_service_mesh.md:36]: Are the quotes around the key mandatory?

File: docs/virtual_machines/istio_service_mesh.md
Comment (phoracek) [inline docs/virtual_machines/istio_service_mesh.md:28]: Instead of "webhook to injext the proxy container" we could say "to register the VM into the service mesh" or something similar? I'm thinking how to make this closer to an end user who does not care about proxy containers being involved

File: docs/virtual_machines/istio_service_mesh.md
Comment (phoracek) [inline docs/virtual_machines/istio_service_mesh.md:39]: Is this needed?

File: docs/virtual_machines/istio_service_mesh.md
Comment (phoracek) [inline docs/virtual_machines/istio_service_mesh.md:52]: Is this needed?

**Review by phoracek — CHANGES_REQUESTED** [review]
Overall it looks good, some suggestions

File: docs/virtual_machines/istio_service_mesh.md
Comment (rhrazdil) [reply docs/virtual_machines/istio_service_mesh.md:52]: Nope

File: docs/virtual_machines/istio_service_mesh.md
Comment (rhrazdil) [reply docs/virtual_machines/istio_service_mesh.md:36]: `true` must be quoted, the key does not

File: docs/virtual_machines/istio_service_mesh.md
Comment (rhrazdil) [reply docs/virtual_machines/istio_service_mesh.md:110]: Ack, added  the `Troubleshooting` section

File: docs/virtual_machines/istio_service_mesh.md
Comment (rhrazdil) [reply docs/virtual_machines/istio_service_mesh.md:81]: Added

File: docs/virtual_machines/istio_service_mesh.md
Comment (rhrazdil) [reply docs/virtual_machines/istio_service_mesh.md]: Moved this to Prerequisites section and also added a case to the Troubleshooting section describing what happens when the istio-cni NAD is missing

File: docs/virtual_machines/istio_service_mesh.md
Comment (rhrazdil) [reply docs/virtual_machines/istio_service_mesh.md]: Updated accordingly to your suggestion

File: docs/virtual_machines/istio_service_mesh.md
Comment (rhrazdil) [reply docs/virtual_machines/istio_service_mesh.md]: Neither of these is strictly needed, only recommended.
Removed the `version` label. I'm keeping the app label as the service is referring to it.
I'm wondering if we should provide the service.... I'm keeping it there for now

File: docs/virtual_machines/istio_service_mesh.md
Comment (maiqueb) [inline docs/virtual_machines/istio_service_mesh.md:3]: s/microservices/pods/ ? 

My point is that a VM runs in a pod - and is not a microservice.

File: docs/virtual_machines/istio_service_mesh.md
Comment (maiqueb) [inline docs/virtual_machines/istio_service_mesh.md:6]: ipv4 only ? Am I wrong ?

File: docs/virtual_machines/istio_service_mesh.md
Comment (maiqueb) [inline docs/virtual_machines/istio_service_mesh.md:79]: You've already mentioned this twice - once in the `Prerequisites` section, another on the "Create a VirtualMachineInstance with enabled Istio proxy injecton". I would remove it from the latter, or here.

File: docs/virtual_machines/istio_service_mesh.md
Comment (maiqueb) [inline docs/virtual_machines/istio_service_mesh.md:83]: If you're mentioning this istio tool, you should - imo - add it to the `prerequirements` section - link how to install it or something alike.

File: docs/virtual_machines/istio_service_mesh.md
Comment (maiqueb) [inline docs/virtual_machines/istio_service_mesh.md:94]: Would it make sense to briefly explain what these `CDS / LDS / EDS / RDS` entries are ? Or point to istio's docs.

I see they're synched, that seems good, but I don't understand what they are. 

Maybe I'm wrong.

**Review by maiqueb — COMMENTED** [review]
Minor nits. The guide is simple, and informative. Nice.

File: docs/virtual_machines/istio_service_mesh.md
Comment (rhrazdil) [reply docs/virtual_machines/istio_service_mesh.md:3]: That's a good point

File: docs/virtual_machines/istio_service_mesh.md
Comment (rhrazdil) [reply docs/virtual_machines/istio_service_mesh.md:6]: Correct! Thanks, I missed that

File: docs/virtual_machines/istio_service_mesh.md
Comment (rhrazdil) [reply docs/virtual_machines/istio_service_mesh.md:79]: In prerequisites section I'm mentioning the namespace label, this is a VMI annotation, it's a different thing.

To explaing this more broadly, I'm mentioning it in the note on purpose, because for regular pods, having the annotation is not required, labelled namespace is enough to get the sidecar proxy injected. We use the annotation to perform special configuration, which is why we always need it.
(This feature gap doesn't exist on d/s because d/s service mesh also requires each workload to be annotated explicitly)

Do you still think that having the Note is unnecessary?

Comment (maiqueb) [reply docs/virtual_machines/istio_service_mesh.md:79]: Oh I see. That changes the picture.

Would you make this note a bit more specific ? Mention "Each Istio aware VMI must feature the ...." or something like that. 

I honestly missed the point this was on the VMI.

File: docs/virtual_machines/istio_service_mesh.md
Comment (rhrazdil) [reply docs/virtual_machines/istio_service_mesh.md:94]: I think it  makes sense to point to the appropriate place in Istio [docs](https://istio.io/latest/docs/ops/diagnostic-tools/proxy-cmd/) . Unfortunately, these shortcuts are not explained in the istio docs ATM, I have filed an issue asking for documenting them: https://github.com/istio/istio/issues/34139

File: docs/virtual_machines/istio_service_mesh.md
Comment (rhrazdil) [reply docs/virtual_machines/istio_service_mesh.md]: Yep, your suggestion sounds more clearly. Updated

File: docs/virtual_machines/istio_service_mesh.md
Comment (rhrazdil) [reply docs/virtual_machines/istio_service_mesh.md:97]: Done

File: docs/virtual_machines/istio_service_mesh.md
Comment (rhrazdil) [reply docs/virtual_machines/istio_service_mesh.md]: Done

File: docs/virtual_machines/istio_service_mesh.md
Comment (maiqueb) [inline docs/virtual_machines/istio_service_mesh.md:18]: nit: this is only required for debugging / troubleshooting. If you have the chance, add that tiny update here.

**Review by maiqueb — APPROVED** [review]
Thank you.

**Review by maiqueb — APPROVED** [review]

File: docs/virtual_machines/istio_service_mesh.md
Comment (phoracek) [inline docs/virtual_machines/istio_service_mesh.md:1]: Looking at the render https://deploy-preview-466--kubevirt-user-guide.netlify.app/virtual_machines/istio_service_mesh/#istio-service-mesh. Istio is now available through search, but not listed in the menu as far as I can tell. Would you add it there via https://github.com/kubevirt/user-guide/blob/ef98ff5d7231e78fa3aaac3120242b0f032a5681/docs/virtual_machines/.pages

File: docs/virtual_machines/istio_service_mesh.md
Comment (phoracek) [inline docs/virtual_machines/istio_service_mesh.md:12]:
```suggestion
- Istio only supports IPv4.
```

File: docs/virtual_machines/istio_service_mesh.md
Comment (phoracek) [inline docs/virtual_machines/istio_service_mesh.md:117]: This renders as 1. again due to it being parsed as a new list: https://deploy-preview-466--kubevirt-user-guide.netlify.app/virtual_machines/istio_service_mesh/#istio-service-mesh

**Review by phoracek — CHANGES_REQUESTED** [review]
Looks good overall

**Comment by phoracek** [comment]
/lgtm
/approve
/hold

Looks very nice! Holding in case @maiqueb  has any notes on changes since his last lgtm.

**Comment by maiqueb** [comment]
/hold cancel

I'm good with it.

**Pull request merged**
